pkg/backup/backup-session.go

```diff
@@ -0,0 +1,345 @@
+package backup
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"time"
+
+	"github.com/appscode/go/log"
+	"github.com/appscode/stash/apis"
+	api "github.com/appscode/stash/apis/stash/v1alpha1"
+	v1beta1_api "github.com/appscode/stash/apis/stash/v1beta1"
+	cs "github.com/appscode/stash/client/clientset/versioned"
+	stash_scheme "github.com/appscode/stash/client/clientset/versioned/scheme"
+	stash_v1beta1_util "github.com/appscode/stash/client/clientset/versioned/typed/stash/v1beta1/util"
+	stashinformers "github.com/appscode/stash/client/informers/externalversions"
+	"github.com/appscode/stash/client/listers/stash/v1beta1"
+	"github.com/appscode/stash/pkg/controller"
+	"github.com/appscode/stash/pkg/eventer"
+	"github.com/appscode/stash/pkg/restic"
+	"github.com/appscode/stash/pkg/status"
+	"github.com/appscode/stash/pkg/util"
+	"github.com/golang/glog"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/runtime"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/tools/cache"
+	"k8s.io/client-go/tools/leaderelection"
+	"k8s.io/client-go/tools/leaderelection/resourcelock"
+	"k8s.io/client-go/tools/record"
+	"k8s.io/client-go/tools/reference"
+	"k8s.io/kubernetes/pkg/apis/core"
+	"kmodules.xyz/client-go/tools/queue"
+)
+
+type Controllers struct {
+	K8sClient            kubernetes.Interface
+	StashClient          cs.Interface
+	MasterURL            string
+	KubeconfigPath       string
+	StashInformerFactory stashinformers.SharedInformerFactory
+	MaxNumRequeues       int
+	NumThreads           int
+	EnableRBAC           bool // rbac for check job
+	ResyncPeriod         time.Duration
+	//backupConfiguratio
+	Name      string
+	Namespace string
+	//Restic
+	SetupOpt     restic.SetupOptions
+	ExtraOpt     util.ExtraOptions
+	BackupOutput restic.BackupOutput
+	BackupOpt    restic.BackupOptions
+
+	Metrics      restic.MetricsOptions
+	//Backup Session
+	bsQueue    *queue.Worker
+	bsInformer cache.SharedIndexInformer
+	bsLister   v1beta1.BackupSessionLister
+
+	Status   status.UpdateStatusOptions
+	Recorder record.EventRecorder
+}
+
+func (c *Controllers) RunBackup() error {
+	stopCh := make(chan struct{})
+	defer close(stopCh)
+
+	//get BackupConfiguration
+	backupConfiguration, err := c.StashClient.StashV1beta1().BackupConfigurations(c.Namespace).Get(c.Name, metav1.GetOptions{})
```

Comment:
do this after getting backup-session event

Reply:
fixed